src/js/wrap.h

```diff
@@ -100,6 +127,51 @@ namespace ccf::js
 
   class Context;
 
+  class JSWrappedAtom
+  {
+  public:
+    JSWrappedAtom() : ctx(NULL), val(JS_ATOM_NULL) {}
+    JSWrappedAtom(JSContext* ctx, JSAtom&& val) : ctx(ctx), val(std::move(val))
+    {}
+    JSWrappedAtom(JSContext* ctx, const JSAtom& value) : ctx(ctx)
+    {
+      val = JS_DupAtom(ctx, value);
+    }
+    JSWrappedAtom(const JSWrappedAtom& other) : ctx(other.ctx)
```

Comment:
Similar to `JSWrappedValue`, it would be good if we could remove this `const&` overload, so that we get compile-time errors if we fail to `move`/`take` an argument.

Reply:
Agreed!